crates/biome_js_analyze/src/lint/style/no_restricted_imports.rs

```diff
@@ -386,115 +860,440 @@ impl ImportRestrictionStatus {
     PartialEq,
     Serialize,
 )]
-#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase", deny_unknown_fields, default)]
-pub struct CustomRestrictedImportOptions {
-    /// The message to display when this module is imported.
-    #[serde(skip_serializing_if = "str::is_empty")]
-    message: Box<str>,
+pub struct PatternOptions {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    group: Option<Box<[Box<str>]>>,
 
-    /// Names of the exported members that should not be used.
-    #[serde(skip_serializing_if = "<[_]>::is_empty")]
-    import_names: Box<[Box<str>]>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    regex: Option<Box<str>>,
 
-    /// Names of the exported members that allowed to be not be used.
-    #[serde(skip_serializing_if = "<[_]>::is_empty")]
-    allow_import_names: Box<[Box<str>]>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    message: Option<Box<str>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    import_names: Option<Box<[Box<str>]>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    allow_import_names: Option<Box<[Box<str>]>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    import_name_pattern: Option<Box<str>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    allow_import_name_pattern: Option<Box<str>>,
```

Comment:
I wonder if we really need these two options in a first version. Do you know if it is widely sued?